### ISSUE TITLE
Svelte 5 functionality issues

### ISSUE THREAD

**Issue by coredoesdev** [vkurko/calendar#359]
Works great in svelte 4. Tried in svelte 5 - this lib is incompatible with the new event system & some buttons don't appear correctly (copy-pasting the demo options, with all plugins installed):
![image](https://github.com/user-attachments/assets/093691ef-557a-43f1-aaa3-8e87dd509ca1)

**Comment by ngDuyAnh** [comment]
Yes, there is some initialize rendering problems with the component in Svelte 5. I had to downgrade to Svelte 4 to use it properly.

**Comment by vkurko** [comment]
Thanks, Svelte 5 support is in the works.

**Comment by ondrejrohon** [comment]
@vkurko need help?

**Comment by vkurko** [comment]
Overall, the experience of #235 can be useful. I don't have a specific plan of action right now (or rather, I initially wanted to rewrite everything from scratch, but it turned out to be too expensive, and I put this idea aside).

So if you can suggest rewriting some part of the code so that it works well in Svelte 5, then please, I'm open to such suggestions.

**Comment by Mathnerd314** [comment]
Well, the SMUI library is an example - first they simply added `<svelte:options runes={false} />` to each file, then they are (still) migrating the files one by one, e.g. in [this commit](https://github.com/hperrin/svelte-material-ui/commit/58edcd2de6145beabf55a72f3494b64f3b82c95a#diff-e7966c8636031425fbae4426251342dfff555a8fa458fa7ebf09fbc3b88e66d8L4), changing those to `runes={true}`. There is an automigrate script `npx sv migrate svelte-5`, but it isn't complete, so doing it incrementally makes more sense IMO.

**Comment by James3UK** [comment]
@vkurko 
The main change required is just renaming the store to something other than state because it clashes with the svelte 5 state rune.
e.g setContext('state'); just needs to be setContext('calendarstate');

**Comment by vkurko** [comment]
Thanks for all the tips. I hope I can devote some time to this task soon.

**Comment by garethdown44** [comment]
Hi! I was just trying out and then found it doesn't seem to work. Just confirming it doesn't support Svelte 5 at all, right? And is it fair to say it won't for some time?

**Comment by vkurko** [comment]
The Svelte 5 version is already in testing. I hope to be able to publish the release in the next week or two.

**Comment by garethdown44** [comment]
Amazing!! Happy to do some early testing if needed when it's ready.

**Comment by vkurko** [comment]
> Amazing!! Happy to do some early testing if needed when it's ready.

Sure. Here is the instruction:

1. There is a new NPM package for the entire library `@event-calendar/svelte`. So you only need to install that single package instead of `@event-calendar/core`, `@event-calendar/time-grid` and so on:

```bash
npm install @event-calendar/svelte
```

2. Then in you Svelte 5 application import and use the calendar and its plugins like this:

```svelte
<script>
import {Calendar, TimeGrid, Interaction} from '@event-calendar/svelte';
</script>

<Calendar plugins={[TimeGrid, Interaction]} options={{events: []}} />
```

The current version is in alpha stage, so please expect bugs.

**Comment by vkurko** [comment]
Does the absence of any feedback mean there are no bugs? 😃

**Comment by kmclane-ten** [comment]
Would you be able to confirm if I should install @event-calendar/svelte as a dependency since this repo's readme recommends installing @event-calendar/core as a dev dependency.

**Comment by vkurko** [comment]
> Is there a GitHub for the new version?

Will be available soon.

> Would you be able to confirm if I should install @event-calendar/svelte as a dependency since this repo's readme recommends installing @event-calendar/core as a dev dependency.

Confirm. For Svelte 5 projects you will need to use repository `@event-calendar/svelte`. The documentation for this is not ready yet.

**Comment by vkurko** [comment]
> Just installed it in a Svelte 5 project as a dev dependency. <Calendar {plugins} {options} /> still renders despite this error. Haven't put it through it's paces yet, but I immediately notice this fixes dayMaxEvents.
> 
> ![Image](https://github.com/user-attachments/assets/48b914b0-bd99-475c-8f6f-96a7e0b71a28)

Thanks for the feedback.

**Comment by vkurko** [comment]
After some testing and experimentation, I decided to continue using the old core package `@event-calendar/core` and not skip the `4.x.x` versions. So please replace `@event-calendar/svelte` with `@event-calendar/core` and install the already available version `4.0.0` which is completely rewritten in Svelte 5. See [changelog](https://github.com/vkurko/calendar/blob/master/CHANGELOG.md#400) and [documentation](https://github.com/vkurko/calendar?tab=readme-ov-file#usage) on the new calendar initialization in ESM/standalone versions.
